### DISCUSSION TITLE
Feedback on first installation and setup

### DISCUSSION THREAD

**Discussion by ovizii** [docmost/docmost#32] [General]
Initially we are asked to created a workspace upon first login:

![image](https://github.com/docmost/docmost/assets/837902/c239648e-6959-4d46-a2c2-ac56b5d8ad20)

Upon login I see:

![image](https://github.com/docmost/docmost/assets/837902/ef5e38f4-cb7b-4aa7-acbe-753cd01ae669)

The question is now, how do spaces and workspaces relate to each other? It looks like spaces are part of a workspace? Would love to hear more details about how this was meant to work, maybe there are better words out there to describe the situation and planned use?

![image](https://github.com/docmost/docmost/assets/837902/c7af4b38-e41d-4b29-b004-521f6a12d641)

Found this. Does this mean we can use spaces like teamA, teamB or projectA, projectB? 
Then what do we call a workspace? Do we use the Company name for the workspace? It looks like the workspace is unique per docmost instance?

![image](https://github.com/docmost/docmost/assets/837902/cbff1986-5e91-4adc-b55b-0c1980c073c1)

I'm not complaining just thinking out loud, maybe it gives you some inspiration for the manual :-)

**Comment by ovizii** [comment]
Also, I can't seem to find where to delete the default space called "general"

**Reply by Philipinho** [reply]
In the current release, there is no UI for deleting Spaces unless via the Database. It will come soon.
Thank you for trying it out. I appreciate it.

**Reply by ovizii** [reply]
Thanks and no worries, I saw your post on reddit/selfhosted and realized it's the initial release of this tool.

**Reply by acureau** [reply]
I had this same question about deleting the general space, I'd have just renamed it but it seems that the stub is permanently set to general. I will look into removing it via the database. Thanks for your response and your effort on this project. It's in a seriously impressive state for an initial release!

Update:
For anyone looking to do this, follow these steps (at your own risk).

1. Run `docker ps -a`, copy the container ID of `docmost-db-1`.
2. Run `docker exec -it <CONTAINER ID> psql -U docmost` to enter psql shell.
3. Run the query `select * from spaces;`, copy the ID of the general space.
4. Run the query `delete from spaces where id = '<SPACE ID>';`.

Once again, at your own risk! I don't know if this will break something but it seems to work fine on my install.

**Reply by Philipinho** [reply]
@acureau Thank you for your kind words, it means much.
I left out the slug changing so I can best figure out how internal links should work. I will make it editable once that aspect is covered.

**Comment by Philipinho** [comment]
I will commence work on the user guide documentation soon.

### Hierarchy Overview
Workspace -> Space -> Page

### Definitions
**Workspace:** The parent organization that contains Spaces.
**Space:** A sub-unit within a Workspace, containing Pages.
**Page:** Individual documents within a Space.

### Roles and Permissions
In a Workspace, users can have the following roles:
Owner
Admin
Member

### Role Capabilities
**Owners and Admins:** Can create Spaces.
**Members:** Cannot create Spaces.

### Space Creation and Administration
When a Space is created, the creator automatically becomes the Space Admin, granting them full access to the Space.
Space Admins can add other users to the Space.

You can create Spaces for various purposes, such as a space for the product team, a space for frontend developers, or a dedicated space for a new company project.

### Space Permissions
Spaces have three permission levels:

- Full Access
- Can Edit
- Can View

### Space Permission Details
**Full Access:**
- Edit Space information.
- Manage Space members.
- Create and manage Pages in the Space.
- Can re-order sidebar pages

**Can Edit:**
- Read Space information.
- See other Space members.
- Create and manage Pages in the Space.
- Can re-order sidebar pages
- Cannot add new members.

**Can View:**
- Has read-only access to the Space.
- View Space information and members.
- View Pages without the ability to edit, delete, or share them.
- Cannot re-order sidebar pages

### Granting Permissions
Admins can grant any of the three permissions to users within a Space, either directly or via a Group.
Ordinary Workspace members can become Space Admins (Full access).

> [!NOTE]
> If a user belongs to a Space through multiple means (e.g., via a group and directly), the membership with the highest permission level will take precedence.

### Pages
**Pages:** Pages belong to Spaces and inherit the permissions of the Space.
For example, if a user has "Can View" permission in a Space, they can only view the Pages within that Space but cannot edit them.
You cannot grant individual permissions to pages (at least, not yet).

**Reply by itst** [reply]
Yay! Thank you!

Can a installation contain more than one workspace, or as an installation a workspace?

**Reply by Philipinho** [reply]
Hi @itst, multi-workspace is reserved for the cloud version.

**Reply by pedroponte** [reply]
> Hi @itst, multi-workspace is reserved for the cloud version.

I am testing docmost for privacy reasons on premise, as an alternative to notion which is cloud based.
I'd love to have the multi workspace feature available in the self hosted version also, even if a license is required.
TY for this project.
I love the diagram features in particular.

**Comment by Philipinho** [comment]
You can now delete Spaces in v0.4.
